src/main/java/org/mskcc/smile/service/impl/ValidRequestCheckerImpl.java

```diff
@@ -391,9 +414,14 @@ public Boolean hasRequestId(String json) throws JsonProcessingException {
         return (!isBlank(requestId));
     }
 
-    private Boolean hasIgoId(Map<String, Object> sampleMap) {
+    private String getPrimaryOrIgoId(Map<String, Object> sampleMap) {
         Object igoIdOrPrimaryId = ObjectUtils.firstNonNull(sampleMap.get("igoId"),
                sampleMap.get("primaryId"));
+        return igoIdOrPrimaryId == null ? null : igoIdOrPrimaryId.toString();
```

Comment:
Does an empty string "" in json resolve to a null value or an empty string?  Do we need to check for isBlank here?

Reply:
that's a better check - I'll update